src/scripts/arabic.rs

```diff
@@ -0,0 +1,140 @@
+const NONE: u32 = u32::MAX;
```

Comment:
I think there are Rust idioms that might come in handy here, e.g. `Option<T>`.

Reply:
Thanks for the suggestion 🙏 
But if I adopt the Option<T> idiom, will I need to change the type of the array to the following?
```rs
static ARABIC_FORMS: [[Option<u32>; 6]; 42] = [
```

If yes, is it still preferable given that the optional codes start from index 2 in the inner array:
```rs
  [ 0xFE80, 0xFE80, NONE, NONE, NONE, NONE],
```

Because isolated and final forms exist for all characters except the placeholders which are made `0` in this case.